### DISCUSSION TITLE
Fields Visibility & Filtering Many to one options

### DISCUSSION THREAD

**Discussion by bernatvadell** [directus/directus#6760] [Feature Requests]
Hi mates!

I'm starting my second project with Directus, to put it in context, it is a platform where you can manage properties (a real estate agency).

These properties are categorized with a type and based on this, some fields or others are shown.

What I want to propose with a practical example is the following: I have a property type selector, when the "garage" type is selected I will show options such as, Is the door automatic? Type of space (car, motorcycle, etc. .), ... However, if a "house" is selected, the options to display will be "appliances?", "Number of rooms", etc ...

I would like to collaborate with Directus and try to implement this functionality (if it doesn't already exist).

Obviously, you are the ones who know the platform best and you can tell me how to plan the development.

In my head, I was thinking if it would be interesting to be able to define a "filter" in the definition of the schema of a field, in which you could indicate when the field can "operate".

The only thing left to know, is if the filter processing engine can be reused on the client side, to make the change reactive and what would be the appropriate place to store this new field, at first glance in the directus_fields table?

On the other hand, I need to be able to filter an option selector, based on a previously selected value, let's put another practical example:

- I have a selector with categories
- I have a selector (optional) with subcategories of the previously selected category.

Is it possible in many to one relationships to filter the options in relation to a selected filter?

In this case, I was wondering if it would be a good option to include one more configuration parameter in the "many to one" interface. Currently it has "Display Template" and if we add one more parameter called "filter"?

**Comment by rijkvanzanten** [comment]
Heya! We're tracking the conditional fields effort in https://github.com/directus/directus/issues/3151. As a matter of fact, it's the next major feature we'll be working on, so the timing couldn't be better 🙂 Please let me know any and all thoughts on the subject on that ticket 👍🏻

**Comment by bernatvadell** [comment]
Thank you! I'll do so.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
